### PR TITLE
Core/Pet: Gargoyle don't now attacks ghoul target.

### PR DESCRIPTION
**Pull request by HeartWell**
**Changes proposed:**

-  Gargoyle don't now attacks ghoul target.

**Target branch(es):** 1.x.

**Issues addressed:** https://github.com/azerothcore/azerothcore-wotlk/issues/117

**Tests performed:** (Does it build, tested in-game, etc)

- Tested and working

### CODE REVIEW

**Comment by sucofog** [comment]
Working as intended for me

**Comment by ShinDarth** [comment]
thanks @sucofog for the feedback, the PR is almost ready to be merged.

@HeartWell  can you check my comments please?

**Comment by HeartWell** [comment]
@ShinDarth comments?

**Comment by ShinDarth** [comment]
@HeartWell yes, I started a review: you can check it in the [diff of the PR](https://github.com/azerothcore/azerothcore-wotlk/pull/229/files)

**Comment by HeartWell** [comment]
it's empty)

**Comment by ShinDarth** [comment]
@HeartWell sorry it was my fault, I started a review but I forgot to submit it. Now I did it, so you can see my comments.

**Comment by HeartWell** [comment]
@ShinDarth https://github.com/azerothcore/azerothcore-wotlk/commit/4da34f630f667c8900dcf58b7f64de46740475cb i revert this commit.

File: src/scripts/Pet/pet_dk.cpp [hunk 1]
```diff
@@ -75,68 +75,34 @@ class npc_pet_dk_ebon_gargoyle : public CreatureScript
...
-                    
-                    if (dkTarget && dkTarget != me->GetVictim() && me->IsValidAttackTarget(dkTarget))
+                    Unit* selection = owner->ToPlayer()->GetSelectedUnit();
+                    if (selection && selection != me->GetVictim() && me->IsValidAttackTarget(selection)) 
                     {
                         me->GetMotionMaster()->Clear(false);
-                        SwitchTargetAndAttack(dkTarget);
-                        return;
+                        SetGazeOn(selection);
                     }
```
Comment (HeartWell) [inline src/scripts/Pet/pet_dk.cpp]: Here all right, if the first 'if' wrong, then check the following.

File: src/scripts/Pet/pet_dk.cpp [hunk 2]
```diff
@@ -193,28 +159,30 @@ class npc_pet_dk_ebon_gargoyle : public CreatureScript
...
                         if ((*iter)->GetAura(SPELL_DK_SUMMON_GARGOYLE_1, me->GetOwnerGUID()))
                         {
-                            SwitchTargetAndAttack((*iter));
+                            (*iter)->RemoveAura(SPELL_DK_SUMMON_GARGOYLE_1, me->GetOwnerGUID());
+                            SetGazeOn(*iter);
+                            _targetGUID = (*iter)->GetGUID();
                             break;
                         }
                 }
-
```
Comment (HeartWell) [inline src/scripts/Pet/pet_dk.cpp]: Why after each 'if' new line if all in one unit.

File: src/scripts/Pet/pet_dk.cpp [hunk 3]
```diff
@@ -193,28 +159,30 @@ class npc_pet_dk_ebon_gargoyle : public CreatureScript
...
                     _despawnTimer -= diff;
+                    if (!UpdateVictimWithGaze())
+                    {
+                        MySelectNextTarget();
+                        return;
+                    }
 
                     _initialCastTimer += diff;
                     _selectionTimer += diff;
-
```
Comment (HeartWell) [inline src/scripts/Pet/pet_dk.cpp]: Why after each 'if' new line if all in one unit.

**Pull request merged**